### PR TITLE
Allow manifest compaction to generate multiple content pieces

### PR DESCRIPTION
**Pull request by ashmrtn**
Allow manifest compaction to generate multiple content pieces. This currently uses number of manifests in a content piece as a proxy for the total size the content piece will be. The number of manifests per content piece is a number out of hat that's meant to be high enough to not trigger the creation of multiple content pieces in most cases while still hopefully avoiding errors about content pieces being too large

This change is only enabled during compaction to reduce blast radius and because compaction already disables content manager flushes to ensure content writes and deletions are seen by other clients atomically (at least assuming no other content manager flushes if this returns an error)

(for completeness)
Even if content writes aren't atomic, then I think things would still work out alright if there was concurrent compaction. For example, the following could occur if content writes didn't appear atomically:
1. client A begins compaction
2. client A write content piece 1
3. client B refreshes content manager and sees content piece 1
4. client B begins compaction
5. client B writes content pieces 3 and 4
6. client B deletes old content pieces, including content piece 1
7. client A writes content piece 2
8. client A deletes old content pieces. It doesn't know about content pieces 3 and 4 because the manifest manager hasn't been refreshed since compaction started

The end result would be partial duplication of manifest data as content piece 2 would contain some info present in content pieces 3 and 4. However, content pieces 3 and 4 should still have all manifest entries present in the now-deleted content piece 1 as content piece 1 was included in client B's compaction (that's why it was deleted)

### CODE REVIEW

File: repo/manifest/committed_manifest_manager.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 	"github.com/kopia/kopia/repo/content/index"
 )
 
+const maxManifestsPerContent = 1000000
```
Comment (pandeyabs) [inline repo/manifest/committed_manifest_manager.go:21]: Curious, how did we get to this number? Is this assuming some average size per manifest entry from data we have?

File: repo/manifest/manifest_manager_test.go [hunk 2]
```diff
@@ -495,3 +495,71 @@ func getManifestContentCount(ctx context.Context, t *testing.T, mgr *Manager) in
...
+
+	foundContents = getManifestContentCount(ctx, t, mgr)
+	assert.Equal(t, 7, foundContents)
+
+	// Run compaction which should result in multiple content pieces.
+	err := mgr.Compact(ctx)
+	require.NoError(t, err, "compacting manifests")
+
+	foundContents = getManifestContentCount(ctx, t, mgr)
+	assert.Equal(t, 2, foundContents)
```
Comment (pandeyabs) [inline repo/manifest/manifest_manager_test.go:560]: nice!

**Review by pandeyabs — APPROVED** [review]
Looks good!

File: repo/manifest/committed_manifest_manager.go (hunk 1 above)
Comment (vkamra) [inline repo/manifest/committed_manifest_manager.go:21]: Might be useful (in a follow-up) to make this configurable using an ENV. Allows disabling this feature by setting the value to an arbitrarily high value *and* tuning it to a lower value if needed

**Pull request merged**